stdlib/lang-parser/parserModule/parser.bal

```diff
@@ -0,0 +1,1282 @@
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+import ballerina/log;
+
+type Parser object {
+    //stack which holds the operators during expression rule
+    OperatorStack oprStack = new();
+    //fill the operator map
+    OperatorMapper oprMap = new();
+    //fill value type map
+    ValueTypeMapper valMap = new();
+    //stack which holds the expression for the expression rule
+    ExprStack expStack = new();
+    //keeps track on error recovery
+    boolean errorRecovered = true;
+    //keep track of error tokens
+    Token[] errTokens = [];
+    //error token count
+    int errCount = 0;
+    //keeps track on next expected token should be an operator or operand
+    boolean expOperand = true;
+    //if invalid occurence is captured in an expression
+    boolean invalidOccurence = false;
+    // tuple list expression count
+    int tupleListPos = 0;
+    //counter in the operator token in list in tuple literal expression
+    int separatorCount = 0;
+    //there is a operator in the prior
+    boolean priorOperator = false;
+
+
+    private ParserBufferReader parserBuffer;
+
+    public function __init(ParserBufferReader parserBuffer) {
+        self.parserBuffer = parserBuffer;
+    }
+
+    public function parse() returns PackageNode {
+        DefinitionNode?[] dList = [];
+        //definition list position
+        int pos = 0;
+        //parse while the current token is not EOF token
+        while (!self.parserBuffer.isEOFToken()) {
+            Token currToken = self.parserBuffer.consumeToken();
+
+            if (currToken.tokenType == FUNCTION) {
+                FunctionNode function1 = self.parseFunction(currToken);
+                DefinitionNode defNode = function1;
+                dList[pos] = defNode;
+                pos += 1;
+            }
+        }
+        //consume the EOF Token
+        Token currToken = self.parserBuffer.consumeToken();
+        PackageNode pkNode = {
+            nodeKind: PACKAGE_NODE,
+            tokenList: [currToken],
+            definitionList: <DefinitionNode[]>(dList.freeze())
+        };
+        return pkNode;
+    }
+
+    # Method to consume the token from the token buffer if the token matches the expected token.
+    # If the token doesnt match , panic recovery method will be invoked.
+    # 
+    # +mToken - token type.
+    # +rule - grammar rule to which the token should be parsed.
+    # +return - Token.
+    function matchToken(int mToken, NodeKind rule) returns Token {
+        if (self.LAToken(1) == mToken) {
+            Token currToken = self.parserBuffer.consumeToken();
+            return currToken;
+        } else {
+            if (self.LAToken(1) == LEXER_ERROR_TOKEN) {
+                log:printError(self.LookaheadToken(1).lineNumber + ":" + self.LookaheadToken(1).startPos + ": expected " +
+                tokenNames[mToken] + ";found an invalid token '" + self.LookaheadToken(1).text + "'");
```

Comment:
Better to update the indentation since this line is a continuation of the last line. Also update line 94 as well.